### PR TITLE
Add time command

### PR DESCRIPTION
**Pull request by rjoleary**
The output format abides to the POSIX `-p` argument specified here:
http://pubs.opengroup.org/onlinepubs/9699919799/utilities/time.html

The test could not be made completely deterministically as different
running times may result in different output. A regex match ensures the
output is accurate to within a reasonable degree of accuracy.

### CODE REVIEW

File: cmds/time/time.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	// Check for io erros.
+	exitErr, ok := err.(*exec.ExitError)
+	if err != nil && !ok {
+		log.Fatal(err)
+	}
+
+	// Print usage.
+	printTime("real", real)
+	printTime("user", cmd.ProcessState.SystemTime())
+	printTime("sys", cmd.ProcessState.UserTime())
```
Comment (rjoleary) [inline cmds/time/time.go]: Oops, I swapped these two by accident.

Comment (rjoleary) [inline cmds/time/time.go]: Fixed

**Comment by rjoleary** [comment]
I just noticed. It is the same on my Ubuntu machine. There is a bash builtin `time` and a `/usr/bin/time`. Is there any particular reason both are needed?

**Comment by rjoleary** [comment]
It looks like they both have drawbacks:

- External `time` cannot measure the time required by a builtin.
- Builtin `time` can measure builtins, but cannot measure user and sys of those builtins (goroutines can be rescheduled among kernel threads which breaks `RUSAGE_THREAD` measurement).

User and sys are important to measure how much time is spent doing work or being blocked.

So, to fix this PR, should I do all these?

- Add the man-page to rush/time.go
- Add the measurement and printing of real and sys times to rush/time.go
- Move the test to rush/time_test.go

Also, how are builtins tested?

**Comment by rminnich** [comment]
Yeah, I recall I put the time command as a builtin so I could time stuff like builtins. You can even do things like
time time time time time date
to see the overhead on each step.

your fixes look fine. We don't test builtins, in fact I never even thought about it.

**Comment by rjoleary** [comment]
Sounds good. I also added a test! The user/sys measurements are only shown if the command is not builtin:

```
% time cd .
real 0.000
% time time time time cd .
real 0.000
real 0.000
real 0.000
real 0.000
% time ls
cd.go  exit.go	parse.go  rush	rush.go  rush_test.go  time.go
real 0.002
user 0.000
sys 0.002
```

**Pull request merged**